src/main/java/uk/gov/hmcts/darts/datamanagement/service/impl/InboundToUnstructuredProcessorImpl.java

```diff
@@ -53,24 +50,47 @@ public class InboundToUnstructuredProcessorImpl implements InboundToUnstructured
             FAILURE_EMPTY_FILE.getId()
         );
 
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final InboundToUnstructuredProcessorSingleElement singleElementProcessor;
+
+    @Value("${darts.automated.task.inbound-to-unstructured.threads:20}")
+    private final Integer threads;
+
+
     @Override
-    public void processInboundToUnstructured() {
+    public void processInboundToUnstructured(int batchSize) {
         log.debug("Processing Inbound data store");
-        processAllStoredInboundExternalObjectsOneCall();
-    }
-
-    private void processAllStoredInboundExternalObjectsOneCall() {
         List<ExternalObjectDirectoryEntity> inboundList = externalObjectDirectoryRepository.findEodsForTransfer(getStatus(STORED), getType(INBOUND),
                                                                                                                 getStatus(STORED), getType(UNSTRUCTURED), 3,
-                                                                                                                limit);
-        int count = 1;
-        for (ExternalObjectDirectoryEntity inboundObject : inboundList) {
-            log.debug("Processing Inbound to Unstructured record {} of {} with EOD {}", count++, inboundList.size(), inboundObject);
-            try {
-                singleElementProcessor.processSingleElement(inboundObject);
-            } catch (Exception exception) {
-                log.error("Failed to move from inbound file to unstructured data store for EOD id: {}", inboundObject.getId(), exception);
-            }
+                                                                                                                batchSize);
+        AtomicInteger count = new AtomicInteger();
+
+        System.out.println("TMP: here 1");
```

Comment:
You might to go through and clear out your System.out statements

Reply:
Updated